business/istio_config.go

```diff
@@ -144,7 +156,18 @@ func (in *IstioConfigService) GetIstioConfigList(criteria IstioConfigCriteria) (
 	go func(errChan chan error) {
 		defer wg.Done()
 		if criteria.IncludeVirtualServices {
-			if vs, vsErr := in.k8s.GetVirtualServices(criteria.Namespace, ""); vsErr == nil {
+			var vs []kubernetes.IstioObject
+			var vsErr error
+			// Check if namespace is cached
+			if kialiCache != nil && kialiCache.CheckIstioResource("VirtualService") && kialiCache.CheckNamespace(criteria.Namespace) {
+				// Cache uses Kiali ServiceAccount, check if user can access to the namespace
+				if _, vsErr = in.businessLayer.Namespace.GetNamespace(criteria.Namespace); vsErr == nil {
```

Comment:
Does it make sense to use them in this order? First CheckIstioResource && CheckNamespace and only after that the user rights are checked?

Reply:
The flow used is:
1) Is cached enabled ?
2) Is VirtualService included in cache ?
3) Is Namespace included in cache ?
4) Can user see Namespace X ?

I guess when using cache, 2) will shortcut if the VS is not included, and 3) and 4) is the most logical order, but probably it can be some cornercase where 4) 3) could have more sense. 
I don't know, this flow makes more sense intially to me.